### ISSUE TITLE
Fail to start server when use s3 express as disk

### ISSUE THREAD

**Issue by zhuo95** [ClickHouse/ClickHouse#69920]
### Question

Error Log
```
 {} <Error> Application: Code: 499. DB::Exception: Message: Unable to parse ExceptionName: IncompleteBody Message: The request body terminated unexpectedly, bucket clickhouse-poc--use1-az6--x-s3, key db/szs/tkbqhssxhmtnflvhrhonatmfxuayj, object size 4: While checking access for disk s3_express: Cannot parse definition from metadata file /var/lib/clickhouse/store/8bb/8bbf012e-dd87-4845-99cf-1ce98f0a8448/asynchronous_metric_log.sql. (S3_ERROR), Stack trace (when copying this message, always include the lines below):

0. DB::Exception::Exception(DB::Exception::MessageMasked&&, int, bool) @ 0x000000000d61323b
1. DB::S3Exception::S3Exception<String const&, String const&, String const&, unsigned long&>(Aws::S3::S3Errors, fmt::v9::basic_format_string<char, fmt::v9::type_identity<String const&>::type, fmt::v9::type_identity<String const&>::type, fmt::v9::type_identity<String const&>::type, fmt::v9::type_identity<unsigned long&>::type>, String const&, String const&, String const&, unsigned long&) @ 0x0000000010543671
2. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<DB::WriteBufferFromS3::makeSinglepartUpload(DB::WriteBufferFromS3::PartData&&)::$_0, void ()>>(std::__function::__policy_storage const*) @ 0x000000001057fbb0
3. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<DB::TaskTracker::add(std::function<void ()>&&)::$_0, void ()>>(std::__function::__policy_storage const*) @ 0x00000000105823c4
4. std::__packaged_task_func<std::function<std::future<void> (std::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, String const&)::'lambda'(std::function<void ()>&&, Priority)::operator()(std::function<void ()>&&, Priority)::'lambda'(), std::allocator<std::function<std::future<void> (std::function<void ()>&&, Priority)> DB::threadPoolCallbackRunnerUnsafe<void, std::function<void ()>>(ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>&, String const&)::'lambda'(std::function<void ()>&&, Priority)::operator()(std::function<void ()>&&, Priority)::'lambda'()>, void ()>::operator()() @ 0x00000000104c1bdc
5. std::packaged_task<void ()>::operator()() @ 0x000000000db705ac
6. ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::worker(std::__list_iterator<ThreadFromGlobalPoolImpl<false, true>, void*>) @ 0x000000000d6cdf7b
7. void std::__function::__policy_invoker<void ()>::__call_impl<std::__function::__default_alloc_func<ThreadFromGlobalPoolImpl<false, true>::ThreadFromGlobalPoolImpl<void ThreadPoolImpl<ThreadFromGlobalPoolImpl<false, true>>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>(void&&)::'lambda'(), void ()>>(std::__function::__policy_storage const*) @ 0x000000000d6d2011
8. void* std::__thread_proxy[abi:v15007]<std::tuple<std::unique_ptr<std::__thread_struct, std::default_delete<std::__thread_struct>>, void ThreadPoolImpl<std::thread>::scheduleImpl<void>(std::function<void ()>, Priority, std::optional<unsigned long>, bool)::'lambda0'()>>(void*) @ 0x000000000d6d0dc3
9. ? @ 0x0000752eab29ca94
10. ? @ 0x0000752eab329c3c

```

config : 

```
<clickhouse>
	<storage_configuration>
	    <disks>
	        <s3_express>
	            <type>s3</type>
	            <endpoint>https://clickhouse-poc--use1-az6--x-s3.s3express-use1-az6.us-east-1.amazonaws.com/db/</endpoint>
	            <region>us-east-1</region>
	             <use_environment_credentials>true</use_environment_credentials>
	        </s3_express>
	    </disks>
	    <policies>
	        <s3_express>
	            <volumes>
	                <main>
	                    <disk>s3_express</disk>
	                </main>
	            </volumes>
	        </s3_express>
	    </policies>
	</storage_configuration>
</clickhouse>
```

**Comment by nvbkdw** [comment]
same error for me, dose anyone has solutions?

**Comment by melvynator** [comment]
Not enough detail. Please reopen if the issue persist.

**Comment by stamsekar** [comment]
Hi, Getting exactly same issue with same traceback while testing clickhouse cluster with s3 express directory bucket as disk. 

Clickhouse version : 2024.12.17

Configured with 
```
            <s3express>
                <type>s3</type>
                <endpoint>https://clickhouse-s3express-uc--aps1-az1--x-s3.s3express-aps1-az1.ap-south-1.amazonaws.com/db/</endpoint>
                <use_environment_credentials>true</use_environment_credentials>
                <region>ap-south-1</region>
            </s3express>
```
```
            <s3express_main>
                <volumes>
                        <main>
                                <disk>s3express</disk>
                        </main>
                </volumes>
            </s3express_main>
```

Getting the same traceback during clickhouse server initialisation.

**Comment by fuziontech** [comment]
Looks like this was reported in another ticket and has a PR merged w/ backport to 24.8 in process 🥳 
https://github.com/ClickHouse/ClickHouse/issues/72078
